viewers/pydicom_PIL.py

```diff
@@ -63,45 +69,149 @@ def get_PIL_image(dataset):
+        im = Image.fromarray(image).convert(mode)
+
+    except:
+        #When pixel data has multiple frames, output the first one
+        im = Image.fromarray(image[0]).convert(mode)
+
+    if not brightness_factor == 1.0:
+        #enhancer will enhance
+        enhancer = ImageEnhance.Brightness(im)
+
+        #save enhanced version
+        better_im = enhancer.enhance(brightness_factor)
+
+        return better_im
+
+    
+    #return default brightness
     return im
 
 
 def show_PIL(dataset):
     """Display an image using the Python Imaging Library (PIL)"""
     im = get_PIL_image(dataset)
+
     im.show()
+
+
+def create_GIF(dataset,brightness_factor = 1.0,filename="clip"):
+    """Create a GIF using the Python Imaging Library (PIL)
+    
+       Only if DICOM file has multiple frames
+
+       Manipulate image brightness using brightness_factor parameter,
+       receives a float value,
+       Default = 1.0
+       Brighter > 1.0 | Darker < 1.0
+
+       Use filename parameter to specify the name of the file created
+       Receives a string value
+    """
+    if not have_PIL:
+        raise ImportError("Python Imaging Library is not available. "
+                          "See http://www.pythonware.com/products/pil/ "
+                          "to download and install")
+
+    if ('PixelData' not in dataset):
+        raise TypeError("Cannot show image -- DICOM dataset does not have "
+                        "pixel data")
+    
+    bits = dataset.BitsAllocated
+
+    samples = dataset.SamplesPerPixel
+
+    if bits == 8 and samples == 1:
+
+        mode = "L"
+
+    elif bits == 8 and samples == 3:
+
+        mode = "RGB"
+
+    elif bits == 16:
+        # not sure about this -- PIL source says is 'experimental'
+        # and no documentation. Also, should bytes swap depending
+        # on endian of file and system??
+        mode = "I;16"
+    else:
+        
+        raise TypeError("Don't know PIL mode for %d BitsAllocated "
+                        "and %d SamplesPerPixel" % (bits, samples))
+
+    #Can only apply LUT if pydicom is installed                    
+    #LUTification returns ndarrays
+    frames = get_LUT_value(dataset)
+
+    #Create the first frame 
+    gif = Image.fromarray(frames[0]).convert(mode)
+
+    if not brightness_factor == 1.0:
+        #enhancer will enhance
+        enhancer = ImageEnhance.Brightness(gif)
+
+        #save enhanced version
+        better_gif = enhancer.enhance(brightness_factor)
+
+        #Create the rest of the frames
+        data = []
+
+        #Loop thru each frame
+        for item in frames[1:]:
+
+            frame = Image.fromarray(item).convert(mode)
+
+            enhancer = ImageEnhance.Brightness(frame)
+
+            better_frame = enhancer.enhance(brightness_factor)
+
+            data.append(better_frame)
+
+        better_gif.save(f'./{filename}.gif', save_all=True, quality=100 , append_images=data, optimize=True, duration=15, loop=0)
+    
```

Comment:
I would suggest not to have the `./` here so user can pass a full file path into the function.  Also would be good to allow `Path` objects, i.e. use `str(filename)` here which would provide the proper pathname if is a `Path`.

Similarly with other save line below.

And ... perhaps can check if the `.gif` is already in the name and only add if necessary.